src/db.js

```diff
@@ -113,15 +113,54 @@ var Database = {
     var defaultUser = config.get('authentication.defaultUser');
     if (defaultUser) {
       var passwordHash = Passwords.hashSync(defaultUser.password);
-      this.db.run('INSERT INTO users (email, password, name) VALUES (?, ?, ?)',
+      this.db.run(
+        'INSERT INTO users (email, password, name) VALUES (?, ?, ?)',
         [defaultUser.email, passwordHash, defaultUser.name],
         function(error) {
-        if (error) {
-          console.error('Failed to save default user.');
+          if (error) {
+            console.error('Failed to save default user.');
+          } else {
+            console.log('Saved default user ' + defaultUser.email);
+          }
+        });
+    }
+
+    // Add any settings provided.
+    var generateSettings = function(obj, baseKey) {
+      const settings = [];
+      
+      for (const key in obj) {
+        let newKey;
+        if (baseKey !== '') {
+          newKey = baseKey + '.' + key;
         } else {
-          console.log('Saved default user ' + defaultUser.email);
+          newKey = key;
         }
-      });
+        
+        if (typeof obj[key] === 'object') {
+          Array.prototype.push.apply(settings,
```

Comment:
`settings = settings.concat(generateSettings(obj[key], newKey))` or `settings.push(...generateSettings(obj[key], newKey))` would be a bit cleaner than push.apply

Reply:
I specifically avoided settings.concat to avoid copying the array unnecessarily. I can switch to spread syntax, though.